modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -58,17 +58,62 @@ const trimBrsFromTableCell = (dom: DOMUtils, elm: Element) => {
+  return Arr.forall(styles, (nodeStyle) => Arr.forall(nonInheritableStyles, (nonInheritableStyle) => !Strings.startsWith(nodeStyle, nonInheritableStyle)));
+};
+
+// Remove children nodes that are exactly the same as a parent node - name, attributes, styles
 const reduceInlineTextElements = (editor: Editor, merge: boolean) => {
   const textInlineElements = editor.schema.getTextInlineElements();
   const dom = editor.dom;
+  // TINY-7326: Figure out what else should be added
+  const shorthandCssProps = [ 'font', 'text-decoration', 'text-emphasis' ];
 
   if (merge) {
-    const root = editor.getBody(), elementUtils = ElementUtils(dom);
+    const root = editor.getBody();
+    const elementUtils = ElementUtils(dom);
 
     Tools.each(dom.select('*[data-mce-fragment]'), (node) => {
-      for (let testNode = node.parentNode; testNode && testNode !== root; testNode = testNode.parentNode) {
-        if (textInlineElements[node.nodeName.toLowerCase()] && elementUtils.compare(testNode, node)) {
-          dom.remove(node, true);
+      const nodeStyleProps = Obj.keys(dom.parseStyle(dom.getAttrib(node, 'style')));
+      if (Type.isNonNullable(textInlineElements[node.nodeName.toLowerCase()]) && hasInheritableStyles(nodeStyleProps)) {
+        for (let parentNode = node.parentNode; Type.isNonNullable(parentNode) && parentNode !== root; parentNode = parentNode.parentNode) {
+          const parentNodeProps = Obj.keys(dom.parseStyle(dom.getAttrib(parentNode, 'style')));
+
+          // Check if the parent has a style conflict that would prevent the child node from being safely removed,
+          // even if a exact node match could be found further up the tree
+          const hasStyleConflict = Arr.exists(nodeStyleProps, (nodeStyleProp) => {
+            // If parent has a longhand property e.g. margin-left but the child (node) style is margin, need to get the margin-left value of node to be able to do a proper comparison
+            // This is is because getting the style using the key of 'margin' on a 'margin-left' parent would give a string of space separated values or empty string depending on the browser
+            const propExists = Arr.exists(parentNodeProps, (parentNodeProp) => parentNodeProp === nodeStyleProp);
+            const isShorthand = Arr.exists(shorthandCssProps, (prop) => nodeStyleProp === prop);
+            if (!propExists && isShorthand) {
+              const longhandProps = Arr.filter(
+                parentNodeProps,
+                (parentNodeProp) => Arr.exists(shorthandCssProps, (prop) => Strings.startsWith(parentNodeProp, prop))
+              );
+              return Arr.exists(longhandProps, (longhandProp) => {
+                const nodeValue = dom.getStyle(node, longhandProp);
+                const parentValue = dom.getStyle(parentNode, longhandProp);
+                return nodeValue !== '' && parentValue !== '' && nodeValue !== parentValue;
```

Comment:
```suggestion
                return Strings.isNotEmpty(nodeValue) && Strings.isNotEmpty(parentValue) && nodeValue !== parentValue;
```

We have a helper function for this, which arguably improves readability.

Reply:
Thanks for the suggestion